lib/js/src/compose/types/chart/gauge.ts

```diff
@@ -36,13 +41,38 @@ export default class GaugeChart extends BaseChart {
     return (d.meta?.steps || []).map(({ label }: any) => label)
   }
 
+  private datasetPostProc (data: any, m: Metric): Number {
+    // Define a valid function to evaluate
+    let fxRaw = (m.fx || defaultFx).trim()
+    if (!fxRaw.startsWith('return')) {
+      fxRaw = 'return ' + fxRaw
+    }
+
+    const fx = new Function('n', fxRaw)
+
+    if (data instanceof Object) {
+      const a = data as TemporalDataPoint
+
+      const n = a.y
+
+      a.y = fx(n)
+    }
+    else {
+      data = fx(data)
+    }
+
+    return data
+  }
+
   makeDataset (m: Metric, d: Dimension, data: Array<number|any>, alias: string) {
     const steps = (d.meta?.steps || [])
 
```

Comment:
Instead of defining the function here i would move the one from chart.ts to util.ts and provide it the correct parameters
Also look at where the datasetPostProc is called in base chart,(start of makeDataset) since you did it after reduce here. The function will receive the data which is an array in all cases.